misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/XMLToRecordConverter.java

```diff
@@ -248,7 +248,9 @@ private static List<Node> getRecordFieldsForXMLElement(Element xmlElement, boole
             if (xmlNode.getNodeType() == org.w3c.dom.Node.ELEMENT_NODE) {
                 Element xmlElementNode = (Element) xmlNode;
                 boolean isLeafXMLElementNode = isLeafXMLElementNode(xmlElementNode);
-                if (!isLeafXMLElementNode || xmlElementNode.getAttributes().getLength() > 0) {
+                if (!isLeafXMLElementNode || xmlElementNode.getAttributes().getLength() > 1
+                        || (xmlElementNode.getAttributes().getLength() == 1
+                                && !XMLNS_PREFIX.equals(xmlElementNode.getAttributes().item(0).getPrefix()))) {
```

Comment:
Please extract this `xmlElementNode.getAttributes()` a variable, and check other places too.